feat: remove big endian

Meyanis95 opened the pull request:
## What is this PR doing?
This PR is to remove the Big Endian interface in Boolify logic.
- Remove the reversing in ValueWire `new_input()` 
- Update `new_constant()` as well
- Update `as_usize` to work with little-endian 

## How can these changes be manually tested?
The original test suit should not be impacted and all the tests should pass.

## Does this PR resolve or contribute to any issues?


## Checklist
- [x] I have manually tested these changes
- [x] Post a link to the PR in the group chat

## Guidelines
- If your PR is not ready, mark it as a draft

voltrevo reviewed (changes requested):
Taking a closer look at this, I realise that there's a bit more going on than "just remove `.reverse()`", as you might have discovered by doing that and seeing the breaking tests.

The tests definitely need updating, since generating circuits with little endian inputs and outputs are different circuits, and a few of the tests check the circuit itself for an exact match.

`eval` also uses big endian encoding and decoding, and needs to be updated.

And in `generate_bristol`:
- We iterate in reverse (search `.rev()`) so that the IDs are generated in big endian order. Simply remove `.rev()` to preserve the little endian order.
- We use the `.last()` bit from each output's `ValueWire` to get the first wire id in the circuit, since `.last()` is the most significant bit. Use `.first()` instead.

Sorry for underestimating this task so much. Going through it has meant figuring out the correct changes myself, but I'm leaving this as an exercise. Alternatively, feel free to reassign this to me and pick a different introductory task. I suggest [this one](https://www.notion.so/pse-team/Deduplicate-shared-code-192d57e8dd7e80efa3e0ffb9b9f1a949?pvs=4) or making your MPC app from the onboarding guide.

Meyanis95 commented:
Thanks a lot for your help 🙏 
I reverted the changes according to your comments.

I rebuilt some of the circuits (`add` and `mul` examples) by hand and checked that the Bristol format generated after the above changes (remove `.reverse()`, `.rev()`, `last()` -> `first()`) was the correct one and assumed all the Bristol representations in the tests needed to be changed according to the new output of the `generate_bristol()` component;    
I also fixed the wires ordering in `eval.rs`.

From
```rust
wires[id_start + j] = (value >> (width - j - 1)) & 1 == 1;
```
To
```rust
wires[id_start + j] = (value >> j) & 1 == 1;
```

All the tests are passing!
